### DISCUSSION TITLE
Using name and code of Ripes

### DISCUSSION THREAD

**Discussion by jvegh** [mortbopet/Ripes#363] [Ideas]
Dear Community,
I am asking this before making anything public. I am quietly using Ripes and I like it. It inspired me to ask 
1/ how about using "Ripes" as part of project's name (it can be a kind of plagiarism or popularization). A project name like "NeuRipes" could express that it is about neurons instead of processors but computationally uses technology of Ripes
2/ how about using code (larger amounts) for a slightly different goal (for education and demonstration, but mainly in biology), of course within its original namespace, sometimes slightly changed; and of course a massive amount of new code in separate namespace

Actually, I want to use the stuff to make simulator of individual biological (and the same method for technical) neurons and their (smaller or larger) systems. I know there are many neuronal simulators, but neither of them operates with temporal  arguments, as follows.

My neurons are time-driven instead of clock driven; but the essential difference is only that the start signal is provided by an external event instead of a central clock edge and similarly the elementary computation step provides the end of computing signal. In this sense it is a simulation of a single-stage processor (with three sub-stages), or several single-stage cooperating processors. Yes, I know that making asynchronous parallel computing technically failed several years ago, using the clock-driven methods and their derivatives, among others, pipelining. However, biological computing systems (and recently: technical ANNs) work in that way and their understanding and simulation is less than limited.
I am using SystemC (using its engine and events, but not electric signals stuff), and only one next_trigger() per elementary units, so I have good hopes to reach simulating several thousands of elementary units on a single PC, and the stuff needed is to handle some numeric functions 10-50 times per elementary operations. It uses the simulated sc_time value to control the operation instead of instruction pointer (von Neumann requested only "proper sequencing" and told that central clocking will stall for fast processors). The simulated and processor times are independent.
The neurons have many inputs (where the data are carried in the arrival times) and one output (where the result is coded in the form of data sending time). The individual elementary units have no real instructions (the assemblies and layers have; but a little different ones); finally our biological neurons are true data-driven systems and the true ANNs attempt to approach that regime. Ripes' tools and facilities could be used to display the internal state parameters and their interdependence in the "single neuron" mode and their cooperation (biological neural assemblies and artificial layers) in the "many-neuron" mode.

Sorry if I chose a wrong category for my unusual question; I found it was the most closely matching one.

Janos

**Comment by mortbopet** [comment]
Hi @jvegh - interesting idea! The answer is quite simple; Ripes is under [an MIT license](https://github.com/mortbopet/Ripes?tab=MIT-1-ov-file#readme) which pretty much allows you to do whatever you want with the source code of Ripes. Second, I'm no lawyer, but I'd assume that since Ripes isn't a trademarked name, you're free to come up with whatever spinoff names you want. In fact, I'd encourage this! There is already, in my opinion, a beautiful and respectful tradition in the open-source community to name new projects in a way that pays homage to their ancestor/parent projects.

**Comment by jvegh** [comment]
OK, thank you. I'm also no lawyer; and that is what I expected.

BTW: its is just more than an idea:
10.3390/informatics8040071
10.1007/s10441-022-09450-6
10.3390/e24081086
